ros_ws/src/arudino/motor/motor.ino

```diff
@@ -73,6 +124,49 @@ ros::NodeHandle nodeHandle;
 ros::Subscriber<std_msgs::Float64> velocitySub(VELOCITY_TOPIC, &velocityCallback);
 
 /**
+ * @brief ROS Publisher for the Encoder Message
+ */
+ros::Publisher encoderPub(ENCODER_TOPIC, &encoderMessage);
+
+void setup() 
+{
+    //setup pins
+    pinMode(FORWARD_PWM_PIN, OUTPUT);
+    pinMode(REVERSE_PWM_PIN, OUTPUT);
+    pinMode(ENABLE_PIN, OUTPUT);
+    pinMode(ENCODER_PIN, INPUT);
+
+    //Set Robot to break when starting
+    digitalWrite(ENABLE_PIN, HIGH);
+    analogWrite(FORWARD_PWM_PIN, 0);
+    analogWrite(REVERSE_PWM_PIN, 0);
+
+    //Setup ROS node and topics
+    node_handle.initNode();
+    node_handle.subscribe(velocitySub);
+    node_handle.advertise(encoderPub);
+
+    //Set the Inturupt on Pin 2
+    attachInterrupt(0, encoderCount, CHANGE);
```

Comment:
Change to:
attachInterrupt(0, encoderCount, RISING);
This will update on the rising edge and we won't have to check if the change was from low to high or not.